src/main/java/uk/gov/pay/ledger/queue/EventMessageHandler.java

```diff
@@ -70,6 +74,15 @@ public void handle() throws QueueException {
         }
     }
 
+    // provides a transactional guarantee, if any of the events fail to process, none of the events will be persisted
+    public void processEventBatch(List<EventMessage> messages) throws QueueException {
+       jdbi.useTransaction(handle -> {
+            for (EventMessage message : messages) {
+                processSingleMessage(message);
+            }
+        });
```

Comment:
Or `jdbi.useTransaction(handle -> messages.forEach(this::processSingleMessage));` 😀

Reply:
I like this a lot but unfortunately the method specifies a thrown exception and java wants you to deal with it. Briefly looking around either you have to do some interface hackery or the scope of the current expression just has it thrown for the scope around the callback.

Reply:
Exceptions are the enemy of lambdas.